src/irmin-pack/inode.ml

```diff
@@ -1570,22 +1583,36 @@ struct
       let v = I.of_inode la ~depth ~length entries in
       Some (Partial (la, v))
 
+    let classify entries =
+      let r = ref [] in
+      for i = Array.length entries - 1 downto 0 do
+        match entries.(i) with None -> () | Some ptr -> r := (i, ptr) :: !r
+      done;
+      match !r with [ (i, ptr) ] -> `Singleton (i, ptr) | l -> `Entries l
+
     let to_elt t =
       let f la (v : _ I.t) =
         if v.stable then `Node (List.of_seq (I.seq la v))
         else
           match v.v with
           | I.Values n -> `Node (List.of_seq (StepMap.to_seq n))
           | I.Tree v ->
-              let entries = ref [] in
-              for i = Array.length v.entries - 1 downto 0 do
-                match v.entries.(i) with
-                | None -> ()
-                | Some ptr ->
-                    let h = I.Ptr.hash la ptr in
-                    entries := (i, h) :: !entries
-              done;
-              `Inode (v.length, !entries)
+              let target = I.Ptr.target ~cache:false ~force:false "to_elt" in
```

Comment:
It I'm not mistake `target` will always raise `Dangling_hash`.

`Node.to_elt` is called during the production of a streamed proof on inodes that were just loaded from the disk. We didn't have time to load their children at this point.

Changing to `~force:true` would read some children that are not going to be included in the proof. It would also have the effect of calling `find` on these blinded inodes, which `find` function is currently wrapped using `with_handler`.

Reply:
I added a test (https://github.com/mirage/irmin/pull/1686/commits/cc5206e688f4b05a06b211f2451e1346c68d2042), indeed it fails with `dangling hash`